src/puya/mir/models.py

```diff
@@ -53,6 +53,30 @@ def __str__(self) -> str:
         return f"byte {format_bytes(self.value, self.encoding)}"
 
 
+@attrs.frozen(eq=False)
+class PushDeployVar(BaseOp):
+    name: str
+    atype: AVMType
+
+    def accept(self, visitor: MIRVisitor[_T]) -> _T:
+        return visitor.visit_push_deploy_var(self)
+
+    @property
+    def op_code(self) -> str:
+        match self.atype:
+            case AVMType.bytes:
+                return "pushbytes"
```

Comment:
I used `pushbytes` and `pushint` here to match beaker - but I'm not sure there's a reason not to just use `bytes` and `int`

Reply:
I believe the reason beaker did this, is because it would allow it to do on chain substitution of template variables in logic signatures. If the `bytes` or `int` pseudo ops were used the values that needed to be replaced would end up in the int and bytes constant blocks. 

I think the current plan is to not support these use cases in puyapy, is that correct @achidlow?